frontend/templates/troubleshooting/index.tsx

```diff
@@ -2,67 +2,64 @@ import { DefaultLayout } from '@layouts/default';
 import { DefaultLayoutProps } from '@layouts/default/server';
 import Head from 'next/head';
 import React from 'react';
-import { Box, Container, Heading } from '@chakra-ui/react';
-import { Problem, TroubleshootingData } from './hooks/useTroubleshootingProps';
-
-const renderStyles = {
-   a: {
-      color: 'blue.500',
-   },
-   'ul, ol': {
-      paddingLeft: 4,
-   },
-};
+import { Box, Container, Flex, Heading } from '@chakra-ui/react';
+import Prerendered from './prerendered';
+import { Section, TroubleshootingData } from './hooks/useTroubleshootingProps';
+import SolutionCard from './solution';
 
 const Wiki: NextPageWithLayout<{
    wikiData: TroubleshootingData;
    layoutProps: DefaultLayoutProps;
 }> = ({ wikiData }) => {
    return (
-      <Container sx={renderStyles}>
-         <Head>
-            <meta name="robots" content="noindex" />
-         </Head>
-         <Heading as="h1">{wikiData.title}</Heading>
-         {wikiData.introduction.map((intro) => (
-            <IntroductionSection key={intro.heading} intro={intro} />
-         ))}
-         {wikiData.solutions.map((solution) => (
-            <SolutionCard key={solution.heading} solution={solution} />
-         ))}
-         {wikiData.conclusion.map((conclusion) => (
-            <ConclusionSection
-               key={conclusion.heading}
-               conclusion={conclusion}
-            />
-         ))}
-      </Container>
+      <Flex
+         direction="column"
+         justifyContent="center"
+         width="100%"
+         fontSize="16px"
+      >
+         <Flex padding="0px 32px 32px" gap="48px" maxW="1280px" id="wrapper">
+            <Flex gap="16px" direction="column" id="main">
+               <Head>
+                  <meta name="robots" content="noindex" />
+               </Head>
+               <Heading as="h1">{wikiData.title}</Heading>
+               {wikiData.introduction.map((intro) => (
+                  <IntroductionSection key={intro.heading} intro={intro} />
+               ))}
+               {wikiData.solutions.map((solution, index) => (
+                  <SolutionCard
+                     key={solution.heading}
+                     index={index + 1}
+                     solution={solution}
+                  />
+               ))}
+               {wikiData.conclusion.map((conclusion) => (
+                  <ConclusionSection
+                     key={conclusion.heading}
+                     conclusion={conclusion}
+                  />
+               ))}
```

Comment:
Minor: I am not a huge fan of mapping in the return.

I tend to pull out these into local variables:

```tsx
const Conclusions = wikiData.conclusion.map((conclusion) => (
                  <ConclusionSection
                     key={conclusion.heading}
                     conclusion={conclusion}
                  />
               ))
               
               
return ( ..... {Conclusions});
```

The main reasons I do this: 1) Reduces the complexity in the return, 2) Easier refactoring if the component logic requires a new functional component

Reply:
I don't love pulling rendered elements into a variable, because I don't seem to see that pattern in the docs much. I'm not 100% opposed, but I'd probably pull them into a component if I wanted to pull them out.